#! /bin/sh /usr/share/dpatch/dpatch-run
## 5_allows_8bit.dpatch by  <ballombe@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Allow %xx in URIs to encode 8bit chars

@DPATCH@
diff -urNad boa-0.94.14rc21~/src/util.c boa-0.94.14rc21/src/util.c
--- boa-0.94.14rc21~/src/util.c	2005-02-22 15:11:29.000000000 +0100
+++ boa-0.94.14rc21/src/util.c	2006-11-28 16:24:06.000000000 +0100
@@ -410,11 +410,6 @@
             uri_old++;
             if ((c = *uri_old++) && (d = *uri_old++)) {
                 *uri = HEX_TO_DECIMAL(c, d);
-                if (*uri < 32 || *uri > 126) {
-                    /* control chars in URI */
-                    *uri = '\0';
-                    return 0;
-                }
             } else {
                 *uri = '\0';
                 return 0;
